File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 1
    
    for k in range(2,number+1):
        gcdN = 0
        for i in range(1,int(k**0.5)+1):
            if k % i == 0: 
                gcdN += 1
                if i != k//i: gcdN += 1
        answer += gcdN if gcdN <= limit else power
        
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let answer = 1;
    
    for (let k = 2; k <= number; k++){
        let gcdN = 0
        for (let i = 1; i <= k ** 0.5; i++){
            if (k % i == 0) {
                gcdN++
                if (i !== k / i) gcdN++;
            };
        }
        answer += (gcdN <= limit? gcdN : power)
    }

    return answer;
}